tests/dragonfly/memory_test.py

```diff
@@ -28,6 +29,12 @@ async def test_rss_used_mem_gap(df_server: DflyInstance, type, keys, val_size, e
     min_rss = 3 * 1024 * 1024 * 1024  # 3gb
     max_unaccounted = 200 * 1024 * 1024  # 200mb
 
+    # There is a big rss spike when this test is ran in one the gh runners (not the self hosted)
+    # and it fails. This rss spike is not observed locally or on our self host runner so
+    # this adjustment is mostly for CI
```

Comment:
It's fine to break this down to another PR, but this to me means we still don't track all stream usage.. no?
It could be something ephemeral which makes it harder to catch locally

Reply:
> but this to me means we still don't track all stream usage. 
I would say maybe because the RSS spike is not necessarily correlated with us not properly tracking memory but also due some pages not being properly released by the allocator. I tried `MEMORY DECOMMIT` but it did not have an effect. The difference in memory on both my `machine and on our own gh runner` is less than 50mb so based on that I would say we do track it pretty accurately. But I could be wrong as I have been many times in the past :smile: